docs/src/@primer/gatsby-theme-doctocat/components/live-preview-wrapper.js

```diff
@@ -1,38 +1,60 @@
-import {BaseStyles, Box, ThemeProvider} from '@primer/components'
+import {BaseStyles, Box, ThemeProvider, useTheme, DropdownMenu, DropdownButton} from '@primer/components'
 import React from 'react'
 
-// This is a temporary way to allow us to preview components in dark mode.
-// We'll replace this component when @primer/components has a first-class
-// API for theme switching.
-function ThemeSwitcher({children}) {
-  const [colorMode, setColorMode] = React.useState('day')
-
-  React.useEffect(() => {
-    function handleKeyDown(event) {
-      // Use `ctrl + cmd + t` to toggle between light and dark mode
-      if (event.key === 't' && event.ctrlKey && event.metaKey) {
-        setColorMode(colorMode === 'day' ? 'night' : 'day')
-      }
-    }
-    document.addEventListener('keydown', handleKeyDown)
-
-    return function cleanup() {
-      document.removeEventListener('keydown', handleKeyDown)
-    }
-  }, [colorMode])
-
-  return <ThemeProvider colorMode={colorMode}>{children}</ThemeProvider>
+function ThemeSwitcher({setShowSwitcher, setIsDropdownOpen}) {
+  const {theme, dayScheme, setDayScheme} = useTheme()
+  const items = Object.keys(theme.colorSchemes).map(scheme => ({text: scheme.replace(/_/g, ' '), key: scheme}))
+  const selectedItem = React.useMemo(() => items.find(item => item.key === dayScheme), [items, dayScheme])
+  return (
+    <DropdownMenu
+      onOpenChange={setIsDropdownOpen}
+      renderAnchor={({children, ...anchorProps}) => (
+        <DropdownButton variant="small" {...anchorProps}>
+          {children}
+        </DropdownButton>
+      )}
+      items={items}
+      selectedItem={selectedItem}
+      onChange={item => {
+        setDayScheme(item.key)
+        // If selected remove itself after few seconds.
+        setTimeout(() => setShowSwitcher(false), 3000)
```

Comment:
I noticed another quirk related to this setTimeout. If you select a theme, then reopen the dropdown menu before the end of the `setTimeout, the button will disappear while the dropdown menu is open:

https://user-images.githubusercontent.com/4608155/135488293-beeab3fd-e1df-417a-bcde-abaf6727e0dd.mp4

There are a couple of ways I think we could fix this:
1. We could remove the `setTimeout` and not worry about hiding the button after selection.
2. We could pass the `open` state to the `ThemeSwitcher` and check that the dropdown isn't open before calling `setShowSwitcher(false)` in the `setTimeout`.

   ```jsx
   function ThemeSwitcher({isDropdownOpen, setIsDropdownOpen, setShowSwitcher}) {
     ...
     return (
       <DropdownMenu
         open={isDropdownOpen}
         onOpenChange={setIsDropdownOpen}
         onChange={() => {
           // If selected remove itself after few seconds.
           setTimeout(() => {
             if (!isDropdownOpen) {
               setShowSwitcher(false)
             }
           }, 3000)
         }
       />
     )
   }
   ```

I don't have a strong opinion about which option we should choose. @pksjce or @mperrotti do you have a preference?

Reply:
🤔 Lets unship that setimeout behaviour. Will keep an eye out for any feedback once this is published.